File: 351004/Leshok/lab3/publisher/src/dtos/CreatorDto.js
class CreatorRequestDto {
  constructor(login, password, firstname, lastname) {
    this.login = login;
    this.password = password;
    this.firstname = firstname;
    this.lastname = lastname;
  }
}

class CreatorResponseDto {
  constructor(id, login, password, firstname, lastname) {
    this.id = Number(id);
    this.login = login;
    this.password = password;
    this.firstname = firstname;
    this.lastname = lastname;
  }
}

module.exports = { CreatorRequestDto, CreatorResponseDto };

File: 351003/Kisel/NoSQL/RV_Kisel_lab3_Task330/Models/Dtos/CreatorDto.cs
using System.ComponentModel.DataAnnotations;

namespace RV_Kisel_lab2_Task320.Models.Dtos;

public class CreatorDto {
    public int Id { get; set; }

    [Required]
    [StringLength(64, MinimumLength = 2)]
    public string Login { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 8)]
    public string Password { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 2)]
    public string Firstname { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 2)]
    public string Lastname { get; set; } = string.Empty;
}